commit fa9b63adabcfa9b724120ef3352cf6fb82b4b9a5
Author: John Johansen <john.johansen@canonical.com>
Date:   Fri Sep 9 16:00:09 2022 -0700

    apparmor: add user namespace creation mediation
    
    Unprivileged user namespace creation is often used as a first step
    in privilege escalation attacks. Instead of disabling it at the
    sysrq level, which blocks its legitimate use as for setting up a sandbox,
    allow control on a per domain basis.
    
    This allows an admin to quickly lock down a system while also still
    allowing legitimate use.
    
    Reviewed-by: Georgia Garcia <georgia.garcia@canonical.com>
    Signed-off-by: John Johansen <john.johansen@canonical.com>

diff --git a/security/apparmor/apparmorfs.c b/security/apparmor/apparmorfs.c
index 6d0848f10ff0..7170349c8af0 100644
--- a/security/apparmor/apparmorfs.c
+++ b/security/apparmor/apparmorfs.c
@@ -2375,6 +2375,7 @@ static struct aa_sfs_entry aa_sfs_entry_mount[] = {
 static struct aa_sfs_entry aa_sfs_entry_ns[] = {
 	AA_SFS_FILE_BOOLEAN("profile",		1),
 	AA_SFS_FILE_BOOLEAN("pivot_root",	0),
+	AA_SFS_FILE_STRING("mask", "userns_create"),
 	{ }
 };